src/utils/screen_utils.py

```diff
@@ -144,10 +144,11 @@ def read_device_state(path_to_device_state: str) -> Dict[str, Any]:
     except json.JSONDecodeError:
         logger.error('JSON Decode Error occurred reading from device state')
         raise json.JSONDecodeError # TODO: Determine error message format
-    saved_timestamp = device_state['saved_timestamp']
+    
+    saved_timestamp = device_state.get('saved_timestamp') if device_state else None
     current_time = time.time()
 
-    if is_at_least_next_day(saved_timestamp, current_time):
+    if not saved_timestamp or is_at_least_next_day(saved_timestamp, current_time):
         device_state['unit_count'] = 0
```

Comment:
Function name is misleading. `read_device_state` implies that the function won't change the state of the device. However, it does change it. You might want to re-think your implementation.

This is my rule of thumb: functions accomplish concepts; a concept conveyed by the function's name. The reason why I say concepts and not tasks is because having a function for each little task would make the code very hard to read. Stick to one concept per function.

Reply:
I agree, good call. I made an attempt. Let me know what you think.